common/src/auth/SharedAccessSignature.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using System.Net;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    public class SharedAccessSignature
+    {
+        protected string _shareAccessSignatureName;
+        protected string _signature;
+        protected string _audience;
+        protected string _encodedAudience;
+        protected string _expiry;
+        protected string _keyName;
+
+        public SharedAccessSignature(string shareAccessSignatureName, DateTime expiresOn, string expiry, string keyName, string signature, string encodedAudience)
+        {
+            if (string.IsNullOrWhiteSpace(shareAccessSignatureName))
+            {
+                throw new ArgumentNullException(nameof(shareAccessSignatureName));
+            }
+
+            ExpiresOn = expiresOn;
```

Comment:
Is this expected to be GMT? If so, we could name it "expiredOnGmt" and make it clear in XML comments.